graveler/graveler.go

```diff
@@ -915,55 +920,70 @@ func newStagingToken(repositoryID RepositoryID, branchID BranchID) StagingToken
 	return StagingToken(v)
 }
 
-func (g *Graveler) CommitExistingMetaRange(ctx context.Context, repositoryID RepositoryID, parentCommitID CommitID, metaRangeID MetaRangeID, committer string, message string, metadata Metadata) (CommitID, error) {
-	repo, err := g.RefManager.GetRepository(ctx, repositoryID)
-	if err != nil {
-		return "", fmt.Errorf("get repository %s: %w", repositoryID, err)
-	}
-
-	if parentCommitID != "" {
-		_, err = g.RefManager.GetCommit(ctx, repositoryID, parentCommitID)
+func (g *Graveler) CommitExistingMetaRange(ctx context.Context, repositoryID RepositoryID, branchID BranchID, parentCommitID CommitID, metaRangeID MetaRangeID, committer string, message string, metadata Metadata) (CommitID, error) {
+	res, err := g.branchLocker.MetadataUpdater(ctx, repositoryID, branchID, func() (interface{}, error) {
+		repo, err := g.RefManager.GetRepository(ctx, repositoryID)
 		if err != nil {
-			return "", fmt.Errorf("get parent commit %s: %w", parentCommitID, err)
+			return "", fmt.Errorf("get repository %s: %w", repositoryID, err)
+		}
+		if parentCommitID != "" {
+			_, err = g.RefManager.GetCommit(ctx, repositoryID, parentCommitID)
+			if err != nil {
+				return "", fmt.Errorf("get parent commit %s: %w", parentCommitID, err)
+			}
 		}
-	}
 
-	ok, err := g.CommittedManager.Exists(ctx, repo.StorageNamespace, metaRangeID)
-	if err != nil {
-		return "", fmt.Errorf("checking for metarange %s: %w", metaRangeID, err)
-	}
-	if !ok {
-		return "", ErrMetaRangeNotFound
-	}
+		branch, err := g.RefManager.GetBranch(ctx, repositoryID, branchID)
+		if err != nil {
+			return nil, err
+		}
+		if branch.CommitID != parentCommitID {
+			return nil, ErrConflictFound
```

Comment:
Can you create a different error here? `Conflict` is over-used and has a different meaning.
Maybe `ErrCommitNotHEADBranch` or any other..